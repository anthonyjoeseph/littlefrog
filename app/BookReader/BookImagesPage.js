// app/BookReader/BookPageImage.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image,
  TouchableHighlight
} from 'react-native';

import SvgUri from 'react-native-svg-uri';
import AImage from 'react-native-image-animation';

class BookImagesPage extends Component{
  constructor(props){
    super(props);
    this.quizButton = this.quizButton.bind(this);
  }
  quizButton(){
    if(this.props.hasButton){
      return (
        <TouchableHighlight onPress={function(){this.props.onQuizSelect()}.bind(this)}>
          <SvgUri width="100" height="100"
                   source={require('./black-rectangle.svg')}  />
        </TouchableHighlight>
      );
    }else{
      return null;
    }
  }

  animations(){
    if(this.props.animations){
      return this.props.animations.map(
        function(animation){
          return (
          <AImage
               key={animation.frames[0]}
               resizeMode='stretch'
               animationRepeatCount= {0}
               animationDuration={ animation.durationMillis }
               animationImages={
                animation.frames.map(
                  function(frame){
                   return ( {uri:frame} );
                  }
                )
               }
               style={
                 {
                   position:'absolute',
                   left: animation.rect.x,
                   bottom: animation.rect.y,
                   width: animation.rect.width,
                   height:animation.rect.height
                 }
               } />
          );
        }.bind(this)
      );
    }else{
      return null;
    }
  }

  render(){
      return (
        <View style={
          {
            width: this.props.width,
            height: this.props.height,
            flexDirection: 'row',
            justifyContent:'flex-start',
            alignItems:'stretch'
          }}>
          <Image
            resizeMode='contain'
            source={this.props.source}
            style={{flex:1, width: null, justifyContent:'center', alignItems:'center'}}>
            {this.quizButton()}
            {this.animations()}
          </Image>
        </View>
      );
  }
}

export default BookImagesPage;
