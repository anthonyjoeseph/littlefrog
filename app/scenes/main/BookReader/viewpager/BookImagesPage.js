// app/BookReader/BookPageImage.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image,
  Platform
} from 'react-native';

import TouchableOutline from './TouchableOutline'

class BookImagesPage extends Component{
  constructor(props){
    super(props);

    this.questionIcons = this.questionIcons.bind(this);
  }
  questionIcons(){
    if(this.props.questionIcons){
      return this.props.questionIcons.map(
        function(questionIcon, questionIndex){
          return (
            <TouchableOutline
              key={questionIndex}
              source={questionIcon.icon.resource}
              width={questionIcon.icon.rect.width * this.props.width}
              height={questionIcon.icon.rect.height * this.props.height}
              left={questionIcon.icon.rect.x * this.props.width}
              bottom={questionIcon.icon.rect.y * this.props.height}
              onPress={
                function(){
                  this.props.onQuizSelect(questionIndex);
                }.bind(this)
              }
              tintColor={'red'}
            />
          );
        }.bind(this)
      );
    }else{
      return null;
    }
  }

  animations(){
    if(this.props.animations){
      return this.props.animations.map(
        function(animation, index){
          var resourceName = animation.resource
          if(Platform.OS === 'ios'){
            resourceName += ".gif";
          }
          return (
            <Image
              key={index}
              source={{uri: resourceName}}
              style={
                {
                  position:'absolute',
                  left: animation.rect.x,
                  bottom: animation.rect.y,
                  width: animation.rect.width,
                  height:animation.rect.height
                }
              }
            />
          );
        }.bind(this)
      );
    }else{
      return null;
    }
  }

  render(){
    if(this.props.shouldRender){
      return (
        <View style={
          {
            width: this.props.width,
            height: this.props.height,
            flexDirection: 'row',
            justifyContent:'flex-start',
            alignItems:'stretch'
          }}>
          <Image
            resizeMode='contain'
            source={this.props.source}
            style={{flex:1, width: null, justifyContent:'center', alignItems:'center'}}>

            {this.animations()}
            {this.questionIcons()}
          </Image>
        </View>
      );
    }else{
      return (
        <View
          width={this.props.width}
          height={this.props.height} />
      );
    }
  }
}

export default BookImagesPage;
