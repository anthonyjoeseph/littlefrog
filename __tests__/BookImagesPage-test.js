// __tests__/BookImagesPage-test.js
import 'react-native';
import React from 'react';
import BookImagesPage from '../app/scenes/main/BookReader/viewpager/BookImagesPage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <BookImagesPage
     source={{uri: "page1"}}
     animations={[
       {
         "rect" :{
           "x" : 120,
           "y" : 120,
           "width" : 70,
           "height" : 60
         },
         "resource" : "spongebob"
       },
       {
         "rect" :{
           "x" : 320,
           "y" : 220,
           "width" : 170,
           "height" : 60
         },
         "resource" : "bored_animation"
       }
     ]}
     questionIcons={[
       {
         "icon" : {
           "rect" :{
             "x" : 0.5,
             "y" : 0.5,
             "width" : 0.15,
             "height" : 0.35
           },
           "resource" : "frog_outline"
         },
         "questions" : [
           {
             "question" : "What is my favorite food?",
             "options" : [
               "Bananas",
               "Razors",
               "Fried Chicken",
               "Pencil Shavings"
             ],
             "correctIndex" : 2
           },
           {
             "question" : "What is my favorite food?",
             "options" : [
               "Bananas",
               "Razors",
               "Fried Chicken",
               "Pencil Shavings"
             ],
             "correctIndex" : 2
           }
         ]
       },
       {
         "icon" : {
           "rect" :{
             "x" : 0.6,
             "y" : 0.6,
             "width" : 0.2,
             "height" : 0.1
           },
           "resource" : "frog_outline"
         },
         "questions" : [
           {
             "question" : "Whddat is my favorite food?",
             "options" : [
               "Banddanas",
               "Razors",
               "Fridded Chicken",
               "Pencil Shavings"
             ],
             "correctIndex" : 2
           }
         ]
       }
     ]}
     onQuizSelect={ () => {}}
     width={500}
     height={500} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
