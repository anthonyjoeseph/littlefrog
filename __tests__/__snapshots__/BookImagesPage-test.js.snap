exports[`test renders correctly 1`] = `
<View
  height={500}
  width={500} />
`;
